react-auth-client/src/sdk/api.js

```diff
@@ -146,13 +152,62 @@ const generateRandomValue = () => {
   return D[0].toString(36);
 };
 
+export const flatten = (objectOrArray, prefix = '', formatter = (k) => (k)) => {
```

Comment:
does lodash have something for this?

Reply:
Nope, lodash doesn't have a key transformation, only [flattenDeep](https://lodash.com/docs/4.17.15#flattenDeep)
Commented as:
```
/**
 * Recursively flattens JSON object with a keys with a prefix parameter and formatted by '_' character
 * Example: from {a: 1, b: {c: 2, d: 3}} to {a: 1, b_c: 2, b_d: 3}
 *
 * @param objectOrArray JSON object to flatten
 * @param prefix a prefix in each flattened object key
 * @param formatter function to make a custom key formatting
 * @returns {*} flattened object
 */

```

Reply:
I wonder if you could use _.transform and maybe another lodash function to achieve this? No big deal though. The comment helps.

Reply:
It is possible, but the LOC or complication won't be less than this variant, I think)
So, if are fine to keep this one, I'll keep it